docs/assignment1.md

```diff
@@ -43,21 +43,23 @@ Based on the commands we agreed to use, we developed all other functional featur
 | ID  | Short name  | Description  |
 |---|---|---|
 | F1  | Commands  | The player can control the main character by typing command-line instructions following this syntax: `command + [target]`, where the (\*) below indicates that a target is optional. <br/> Available `commands` are: <br/><br/> - `move` + [north, east, south, west, area-name] <br/> - `use` + [item] <br/> - `look` + [item]\* <br/> - `interact` + [npc, item] <br/> - `inventory` <br/> - `save` <br/> - `load` <br/> - `quit` <br/> |
+| F1B  | Customizable Commands  | **BONUS:** <br/> Apart from the list of available `commands` mentioned in F1, the modder shall also be allowed to define custom names for commands by editing the game files written in JSON. Additionally, the modder shall be able to add any of the engine supported basic commands to an obstacle or a player. Lastly, It shall also be possible to add parameters to the customizable commands to allow different behaviour. (e.g. `inventory` might display the player's inventory whereas `inventory merchant` might display the merchant's inventory)   |
 | F2  | Movements  | The player may move freely between different areas in four directions as described by the `move` command above, or by specifying a unique area name (optionally given in the description of the current area). Some areas may be blocked by obstacles and therefore inaccessible. Obstacles may be removed by using a specific item in their presence (e.g. "use key" might open a door).  |
 | F3  | Areas | All areas will be represented as nodes in a graph, where edges between nodes indicate two areas are reachable from one another. Each area will have a description so the player can distinguish between different type of areas, however, some areas may not be unique.  |
 | F4  | Obstacles | An obstacle is a special type of object which prohibits movement between two specific areas until a certain item is used in the same area that they inhabit. Having obstacles be generic like this allows some more complex behaviour to exhibit itself - for example, defeating a monster may have them drop an item that "unlocks" an obstacle, therefore inhibiting progress until the monster is defeated. |
```

Comment:
I like this, It's kind of like a hard wall for progress, make sure the player gets some intuition as to how to break through that wall. Not saying it should be spoonfed to them, but at least something.